feat(flutter): update slow and frozen frames and frames delay (#10507)
getsentry/sentry-docs: docs/platforms/flutter/integrations/slow-and-frozen-frames-instrumentation.mdx

---
title: Slow and Frozen Frames and Frames Delay Instrumentation

> description: "Learn more about the Sentry Slow and Frozen Frames and Frames Delay Instrumentation for the Flutter SDK."

Reviewer: ```suggestion
description: "Learn about slow and frozen frames and frame delay and how to track them using Sentry's Flutter SDK."
```

---

Unresponsive UI and animation hitches annoy users and degrade the user experience.

> Three measurements to track these types of experiences are **slow frames**, **frozen frames** and **frames delay**
> Sentry's slow and frozen frames and frames delay instrumentation helps track these measurements.

Reviewer 2: ```suggestion
Use this integration to find out about these problems in your app by tracking and showing you **slow frames**, **frozen frames** and **frames delay** metrics for spans.
```

Reviewer: @kahest hope you don't mind a made a few tweaks to your suggestion :)

> Learn more about frames delay [here](https://develop.sentry.dev/sdk/performance/frames-delay/).

Reviewer: ```suggestion
Learn more about frame delay [here](https://develop.sentry.dev/sdk/performance/frames-delay/).
```

Reviewer: I know it says "frames delay" in our developer docs, but it should be "frame delay" there too. You can say "frame delay" or "frames delays" because "delay" is singular ;p

Author: Yah I was confused about that too, thought it was intentional, thx!

## Instrumentation Behaviour

> The frame metrics are manually calculated with the help of the [addPersistentFrameCallback](https://api.flutter.dev/flutter/scheduler/SchedulerBinding/addPersistentFrameCallback.html) API from the Flutter SDK.

Reviewer 2: ```suggestion
The frame metrics are manually calculated using the [addPersistentFrameCallback](https://api.flutter.dev/flutter/scheduler/SchedulerBinding/addPersistentFrameCallback.html) API from the Flutter SDK.
```

> Frame duration tracking in the Sentry Flutter SDK begins automatically when a span starts, continuously measures each frame's render time until the span finishes, then calculates and attaches frame metrics to the completed span.

Reviewer: ```suggestion
Frame duration tracking in Sentry's Flutter SDK begins automatically when a span starts. It continuously measures each frame's render time until the span finishes, then calculates and attaches frame metrics to the completed span.
```

### Disabling the Instrumentation

> Set `enableFramesTracking` to `false` in the options to disable the instrumentation.

Reviewer 2: it's great that this is now decoupled from routing instrumentation and that users can deactivate it separately from overall tracing 👍
